core/src/main/java/org/infinispan/util/concurrent/locks/OwnableReentrantLock.java

```diff
@@ -40,6 +42,7 @@ protected final Object currentRequestor() {
    }
 
    private void setCurrentRequestor(Object requestor) {
+      assert requestorOnStack.get() == null;
```

Comment:
I don't think this can really happen, and I don't want to read ThreadLocal when I don't have to. Assert does not hurt.